Config loader changes for refactoring recipe

vishwahiremat opened the pull request:
# Description

Adding Environment Configuration Loader. 

- ConfigurationLoader is responsible for getting configurations needed to deploy recipes. It calls UCP to fetch information on environment and application and returns the configuration that represents kubernetes runtime and cloud provider configuration.

## Issue reference

<!--
We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.
-->

https://github.com/project-radius/radius/issues/5093
## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Adds necessary unit tests for change
* [ ] Adds necessary E2E tests for change
* [x] Unit tests passing
* [ ] Extended the documentation / Created issue for it

vishwahiremat committed:
config loader changes

vishwahiremat committed:
addressing review comments

vishwahiremat committed:
fixing nits

vishwahiremat committed:
fix comments

vishwahiremat committed:
addressing comments

vishwahiremat committed:
addressing comments

vishwahiremat committed:
adding unit tests and addressing comments

vishwahiremat committed:
adding comments and resolving pr comments

vishwahiremat committed:
add compute property check

vishwahiremat committed:
adding comments for config struct

ytimocin reviewed (approved):
Please update the docs but other than that LGTM

vishwahiremat committed:
renaming RecipeContext-RecipeMetadata

vishwahiremat committed:
addressing review comments

vishwahiremat committed:
Merge branch 'main' into vishwahiremat/decoupleRecipes

sk593 committed:
Adding Config loader unit tests (#5278)

# Description

Adding unit tests for config loader tests in #5242. Will update based on
changes to that PR and merge into implementation branch. Unit test
changes are located in
[`environment_test.go`](https://github.com/project-radius/radius/blob/c03879381bb0aedc4a4131947ee832df8258dd4d/pkg/recipes/configloader/environment_test.go)
and
[`resoures_test.go`](https://github.com/project-radius/radius/blob/c03879381bb0aedc4a4131947ee832df8258dd4d/pkg/rp/kube/resources_test.go)

## Issue reference

<!--
We strive to have all PR being opened based on an issue, where the
problem or feature have been discussed prior to implementation.
-->

Fixes: https://dev.azure.com/azure-octo/Incubations/_workitems/edit/6076

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of
the following list:

* [ ] Code compiles correctly
* [ ] Adds necessary unit tests for change
* [ ] Adds necessary E2E tests for change
* [ ] Unit tests passing
* [ ] Extended the documentation / Created issue for it

vishwahiremat committed:
addressing comments

vishwahiremat committed:
Merge branch 'main' into vishwahiremat/decoupleRecipes

vishwahiremat committed:
addressing comments

vishwahiremat committed:
fix comment for env loader

vishwahiremat committed:
Trigger Build

vishwahiremat committed:
addressing comments

vishwahiremat committed:
addressing comments

vishwahiremat committed:
Merge branch 'main' into vishwahiremat/decoupleRecipes

vishwahiremat committed:
Merge branch 'main' into vishwahiremat/decoupleRecipes

kachawla reviewed (commented):
As discussed previously, please provide more clarity in the PR description and commit message about the change with a link to the corresponding github issue. Clear commit messages are helpful when someone new is trying to get context about the changes, and it becomes more important for an open source repo.

vishwahiremat committed:
Trigger Build
